pkg/common/config/config.go

```diff
@@ -255,11 +255,18 @@ func (cfg *Config) FromEnv() error {
 // ReadConfig parses vSphere cloud config file and stores it into VSphereConfig.
 // Environment variables are also checked
 func ReadConfig(byConfig []byte) (*Config, error) {
+	var cfg *Config
+
 	if len(byConfig) == 0 {
 		return nil, fmt.Errorf("Invalid YAML/INI file")
 	}
 
-	cfg, err := ReadConfigYAML(byConfig)
+	// Check to see if yaml file before calling parser to prevent unnecessary error messages.
+	isYAML, err := isConfigYaml(byConfig)
```

Comment:
It is better to change this to
```go
if isConfigYaml(byConfig) {
    cfg, err = ReadConfigYAML(byConfig)
   if err != nil {
   // handle error
   }
} else {
   cfg, err = ReadConfigINI(byConfig)
   if err != nil {
   // handle error
   }
}

Reply:
Ok, I like that.  I'll make that change.